lib/logstash/filters/base.rb

```diff
@@ -145,6 +146,24 @@ def filter(event)
     raise "#{self.class}#filter must be overidden"
   end # def filter
 
+
+  # in 1.5.0 multi_filter is meant to be used in the generated filter function in LogStash::Config::AST::Plugin only
+  # and is temporary until we refactor the filter method interface to accept events list and return events list,
+  # just list in multi_filter see https://github.com/elastic/logstash/issues/2872.
+  # refactoring the filter method will mean updating all plugins which we want to avoid doing for 1.5.0.
+  #
+  # @param events [Array<LogStash::Event] list of events to filter
+  # @return [Array<LogStash::Event] filtered events and any new events generated by the filter
+  public
+  def multi_filter(events)
```

Comment:
hmmm, I am not in love with `multi_filter` either but definitely do not like `apply` and do not really like `filter_array`... I don't think it better conveys the purpose. other suggestions?